### PR TITLE
Allow mupen64plus to read GLideN64.custom.ini

### PR DESCRIPTION
**Pull request by loganmc10**
This has nothing to do with refactor_graphics, but I figured I might as well submit it to this branch so I don't cause any merge conflicts.

I have written the mupen64plus part for this as well, but I haven't submitted a PR yet, I wanted to see if this is considered an acceptable solution for this problem before I submitted it to them.

I wouldn't merge this until the other half gets accepted into mupen64plus

### CODE REVIEW

File: src/mupenplus/Config_mupenplus.cpp [hunk 1]
```diff
@@ -187,6 +188,112 @@ bool Config_SetDefault()
...
+	if (ConfigExternalGetParameter == nullptr)
+		return;
+	const char* value;
+	std::string ROMname = RSP.romname;
+	const char* pathName = ConfigGetSharedDataFilepath("GLideN64.custom.ini");
+	for (size_t pos = ROMname.find(' '); pos != std::string::npos; pos = ROMname.find(' ', pos))
+		ROMname.replace(pos, 1, "%20");
+	for (size_t pos = ROMname.find('\''); pos != std::string::npos; pos = ROMname.find('\'', pos))
+		ROMname.replace(pos, 1, "%27");
+	std::transform(ROMname.begin(), ROMname.end(), ROMname.begin(), ::toupper);
```
Comment (loganmc10) [inline src/mupenplus/Config_mupenplus.cpp]: I'm not sure of a better way to convert the spaces into "%20" and the 's into %27, perhaps there is a way?

**Comment by loganmc10** [comment]
The mupen64plus half is here:

https://github.com/loganmc10/mupen64plus-core/commit/a6f1ab3bedaf1daa743c870ff5cebb1c91265075

File: src/mupenplus/Config_mupenplus.cpp (hunk 1 above)
Comment (gonetz) [inline src/mupenplus/Config_mupenplus.cpp]: std::regexp?

**Comment by gonetz** [comment]
It looks ok to me. I'd check how Japanese rom names will work. I could not solve problem of translation such names to QT string on Windows. Result was vary for users with different system locales.

**Comment by Papermanzero** [comment]
Would be great to have this feature. 
This would be perfect, to play different games out-of-the-box without dealing with certain configurations.

**Comment by loganmc10** [comment]
I'm just waiting to hear back from mupen64plus, I submitted a pull request there as well

**Comment by gonetz** [comment]
> I'm just waiting to hear back from mupen64plus

They are not fast. At least they did not say no (yet).

**Comment by loganmc10** [comment]
@gonetz I'm happy to report that this has finally been merged into mupen64plus, this can be merged now.

**Comment by gonetz** [comment]
Great! Merged: eb3c5c07306df

**Pull request closed without merging**